src/deroff.rs

```diff
@@ -1060,23 +1164,56 @@ fn test_comment() {
     assert_eq!(deroffer.s, "\nworld".to_owned());
 }
 
-fn deroff_files(files: &[String]) -> std::io::Result<()> {
+fn deroff_files<P: AsRef<std::path::Path>, S: ToString>(
+    files: &[S],
+    output_dir: P,
+) -> std::io::Result<()> {
+    let output_path = output_dir.as_ref().to_path_buf();
     for arg in files {
+        let arg = arg.to_string();
         eprintln!("processing deroff file: {}", arg);
 
-        let mut file = File::open(arg)?;
+        let mut file = File::open(&arg)?;
         let mut string = String::new();
         if arg.ends_with(".gz") {
             let mut decoder = GzDecoder::new(file);
             decoder.read_to_string(&mut string)?;
         } else {
-            file.read_to_string(&mut string)?;
+            match file.read_to_string(&mut string) {
+                Err(_) => {
+                    // TODO: This is a _bad_ workaround for latin1, we need to correctly decode input files
+                    let mut bytes = Vec::new();
+                    file.read_to_end(&mut bytes)?;
+
+                    for byte in bytes {
+                        string.push(byte as char);
+                    }
+                }
+                _ => {}
+            };
         }
         let mut d = Deroffer::new();
         println!("string: {}", string);
 
         d.deroff(string);
-        d.flush_output(std::io::stdout());
```

Comment:
I thought this is correct?
```python
def deroff_files(files):
    for arg in files:
        sys.stderr.write(arg + "\n")
        if arg.endswith(".gz"):
            f = gzip.open(arg, "r")
            str = f.read()
            if IS_PY3:
                str = str.decode("latin-1")
        else:
            f = open(arg, "r")
            str = f.read()
        d = Deroffer()
        d.deroff(str)
        d.flush_output(sys.stdout)
        f.close()
```
It flushes to `sys.stdout` rather than file.